knowledge-base/grid-loader-during-contextmenu-actions.md

```diff
@@ -0,0 +1,237 @@
+
+````RAZOR
+@using System.Collections.ObjectModel
+
+<TelerikContextMenu @ref="@ContextMenuRef" Data="@MenuItems"
+                    OnClick="@((MenuItem item) => ContextMenuClickHandler(item))">
+</TelerikContextMenu>
+
+<TelerikLoaderContainer Visible="@LoaderVisible" Text="Please wait..." />
+
+<TelerikGrid Data="@GridData" @ref="@GridRef"
+             EditMode="@GridEditMode.Inline"
+             Height="500px"
+             Pageable="true"
+             OnCreate="@CreateItem" OnUpdate="@UpdateHandler"
+             OnRowContextMenu="@OnContextMenu"
+             SelectionMode="@GridSelectionMode.Multiple"
+             @bind-SelectedItems="@SelectedItems">
+    <GridToolBarTemplate>
+        <GridCommandButton Command="Add" Icon="@SvgIcon.Plus">Add Employee</GridCommandButton>
+    </GridToolBarTemplate>
+    <GridColumns>
+        <GridColumn Field=@nameof(SampleData.ID) Editable="false" />
+        <GridColumn Field=@nameof(SampleData.Name) />
+        <GridCommandColumn>
+            <GridCommandButton Command="Save" Icon="@SvgIcon.Save" ShowInEdit="true">Save</GridCommandButton>
+            <GridCommandButton Command="Cancel" Icon="@SvgIcon.Cancel" ShowInEdit="true">Cancel</GridCommandButton>
+        </GridCommandColumn>
+    </GridColumns>
+</TelerikGrid>
+
+@if (SelectedItems.Any())
+{
+    <ul>
+        @foreach (var item in SelectedItems)
+        {
+            <li>@item.Name</li>
+        }
+    </ul>
+}
+
+@code {
+    private ObservableCollection<SampleData> GridData { get; set; }
+    private List<MenuItem> MenuItems { get; set; }
+    private IEnumerable<SampleData> SelectedItems { get; set; } = Enumerable.Empty<SampleData>();
+    private SampleData SelectedPerson { get; set; }
+    private TelerikContextMenu<MenuItem> ContextMenuRef { get; set; }
+    private TelerikGrid<SampleData> GridRef { get; set; }
+    private bool LoaderVisible { get; set; }
+
+    public class MenuItem
+    {
+        public string Text { get; set; }
+        public ISvgIcon Icon { get; set; }
+        public Action Action { get; set; }
+        public string CommandName { get; set; }
+    }
+
+    private async Task OnContextMenu(GridRowClickEventArgs args)
+    {
+        var argsItem = args.Item as SampleData;
+
+        SelectedPerson = argsItem;
+
+        if (args.EventArgs is MouseEventArgs mouseEventArgs)
+        {
+            await ContextMenuRef.ShowAsync(mouseEventArgs.ClientX, mouseEventArgs.ClientY);
+        }
+    }
+
+    private async Task ContextMenuClickHandler(MenuItem item)
+    {
+        if (item.Action != null)
+        {
+            item.Action.Invoke();
+        }
+        else
+        {
+            switch (item.CommandName)
+            {
+                case "BeginEdit":
+                    LoaderVisible = true;
+                    await Task.Delay(3000); //initiates delay
```

Comment:
This implementation may backfire at you, because the developer will replace `await Task.Delay(3000)` with their actual long task and the LoaderContainer will never show. Instead use something like this:

```suggestion
                    await Task.Delay(1); // triggers UI refresh for the LoaderContainer to show
                    await Task.Delay(3000); // replace with the actual long-running task
```